### ISSUE TITLE
Support for specifying ipPools in CNI config

### ISSUE THREAD

**Issue by gunjan5** [projectcalico/cni-plugin#212]
As per the discussion in #152 , first step to accomplish what's listed there is to support CNI config below:
```
{
  type: calico
  ipam: {
    type: calico-ipam,
    ipv4_pools: [<ip-pool-1>, <ip-pool-2>],
    ipv6_pools: [<ip6-pool-1>, <ip6-pool-2>]
  }
}
```

To accomplish this, we'd need to make changes in 2 repos:

1) [libcalico-go](https://github.com/projectcalico/calico-cni)
- First have to make `libcalico-go` IPAM code support accepting multiple ipPools as `AutoAssignArgs`. See https://github.com/projectcalico/libcalico-go/blob/master/lib/client/ipam_test.go#L257 test as an example
- Would have to change https://github.com/projectcalico/libcalico-go/blob/master/lib/client/ipam_types.go#L41 so it can take multiple ipPools
- This is the logic where we honor the ipPool specified: https://github.com/projectcalico/libcalico-go/blob/master/lib/client/ipam_block_reader_writer.go#L61 so that would also need to change along with any other functions using ipPools
- Finally, need to change/write new test cases for this behavior in ipam_test.go

2) [CNI](https://github.com/projectcalico/calico-cni)
- Support loading the new arguments from config: https://github.com/projectcalico/calico-cni/blob/master/ipam/calico-ipam.go and maybe a helper func in the utils package
- Change where it calls `AutoAssign` to pass the list of ipPools: https://github.com/projectcalico/calico-cni/blob/master/ipam/calico-ipam.go#L95
- Finally, modify the tests: https://github.com/projectcalico/calico-cni/blob/master/calico_cni_ipam_test.go

**Comment by lxpollitt** [comment]
cc @caseydavenport @MikeSpreitzer

**Comment by duglin** [comment]
I'll work on trying to do the libcaico-go stuff mentioned in 1.  (finger's crossed :smile: )

**Comment by duglin** [comment]
bullet 1's PR: https://github.com/projectcalico/libcalico-go/pull/271

**Comment by duglin** [comment]
libnetwork-cni PR is: https://github.com/projectcalico/libnetwork-plugin/pull/94
calico-cni PR is: https://github.com/projectcalico/calico-cni/pull/223
cni-plugin docs PR is: https://github.com/projectcalico/calico/pull/275

**Comment by gunjan5** [comment]
Solved by #260
